calcrule_third_party_payment/utils.py

```diff
@@ -11,3 +18,143 @@ def check_bill_exist(instance, convert_to, **kwargs):
             bills = BillItem.objects.filter(line_type=content_type, line_id=claim.id)
             if bills.count() == 0:
                 return True
+
+
+def claim_batch_valuation(work_data):
+    """ update the service and item valuated amount """
+    allocated_contributions = work_data["allocated_contributions"]
```

Comment:
the following (not  allocated_contributions) will have to come from paymentplan params